codeworld-compiler/src/CodeWorld/Compile/Requirements/Eval.hs

```diff
@@ -236,6 +277,24 @@ allDefinitionsOf a m = everything (++) (mkQ [] funcDefs) m ++
         patDefs (PatBind _ pat rhs _) | patDefines pat a = [rhs]
         patDefs _ = []
 
+topLevelNames :: Module SrcSpanInfo -> [String]
+topLevelNames m = everything (++) (mkQ [] names) m
```

Comment:
I don't think this function does what you think it does.  `everything` recursively looks through the entire syntax tree, which means this will also find declarations in `let` and `where` blocks and such.

I actually think that here (and also when looks for type declarations), you don't want to use syb at all!  Instead, just pattern-match on the module constructor and look through that specific list of declarations.  Those are the top-level ones.

Reply:
I think I've fixed this!